packages/react-transition-presence/src/TransitionPresence/TransitionPresence.tsx

```diff
@@ -16,14 +17,21 @@ import { TransitionPresenceContext } from './TransitionPresence.context.js';
 export type TransitionPresenceProps = {
   children: ReactNode;
```

Comment:
Changing this to `ReactElement | null` will make the equality checks a lot easier. You won't have to check for other types in `ReactNode`